middleware/throttle.go

```diff
@@ -2,23 +2,43 @@ package middleware
 
 import (
 	"net/http"
+	"time"
 
 	"github.com/pressly/chi"
 	"golang.org/x/net/context"
 )
 
-// Throttle is a middleware that limits number of currently
-// processed requests at a time.
-func Throttle(limit int) func(chi.Handler) chi.Handler {
-	if limit <= 0 {
+var (
+	defaultTimeout = time.Second * 60
+)
+
+// Throttle is a middleware that limits number of currently processed requests
+// at a time.
+func Throttle(limit int, backloglimit int, timeout time.Duration) func(chi.Handler) chi.Handler {
+	if limit < 1 {
 		panic("middleware.Throttle expects limit > 0")
 	}
 
+	if backloglimit < limit {
+		panic("middleware.Throttle expects backloglimit > limit")
+	}
```

Comment:
Good point, I think it would be better if I change this part and put the backlog number on top of the limit.